client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,301 @@
+        host_name = create_context.host_name
+
+        task_name = instance["task"]
+        folder_path = instance["folderPath"]
+        task_entity = create_context.get_task_entity(folder_path, task_name)
+
+        task_name = task_type = None
+        if task_entity:
+            task_name = task_entity["name"]
+            task_type = task_entity["taskType"]
+        profile = filter_profiles(
+            cls.profiles,
+            {
+                "host_names": host_name,
+                "task_types": task_type,
+                "task_names": task_name,
+                # "product_type": product_type
+            }
+        )
+        if not profile:
+            return []
+        overrides = set(profile["overrides"])
+        if not overrides:
+            return []
+
+        defs = [
+            UISeparatorDef("deadline_defs_starts"),
+        ]
+
+        defs.extend(cls._get_artist_overrides(overrides, profile))
+
+        # explicit
+        defs.append(
+            TextDef(
+                "frames",
+                label="Frames",
+                default="",
+                tooltip="Explicit frames to be rendered. (1, 3-4)"
+            )
+        )
+
+        defs = cls._host_specific_attr_defs(create_context, instance, defs)
+
+        defs.append(
+            UISeparatorDef("deadline_defs_end")
+        )
+
+        return defs
+
+    @classmethod
+    def _get_artist_overrides(cls, overrides, profile):
+        """Provide list of Defs that could be filled by artist"""
+        # should be matching to extract_jobinfo_overrides_enum
+        default_values = {}
+        for key in overrides:
+            default_value = profile[key]
+            if isinstance(default_value, list):
+                default_value = ",".join(default_value)
+            default_values[key] = default_value
+
+        attr_defs = [
+            NumberDef(
+                "chunk_size",
+                label="Frames Per Task",
+                default=default_values.get("chunk_size"),
+                decimals=0,
+                minimum=1,
+                maximum=1000
+            ),
+            NumberDef(
+                "priority",
+                label="Priority",
+                default=default_values.get("priority"),
+                decimals=0
+            ),
+            TextDef(
+                "department",
+                label="Department",
+                default=default_values.get("department")
+            ),
+            TextDef(
+                "group",
+                label="Group",
+                default=default_values.get("group")
+            ),
+            TextDef(
+                "limit_groups",
+                label="Limit Groups",
+                # multiline=True,  TODO - some DCC might have issues with storing multi lines
+                default=default_values.get("limit_groups"),
+                placeholder="machine1,machine2"
+            ),
```

Comment:
This should really be a multiselection enum.

Also, artist should be able to pick whether this is allow list or deny list.

Plus, I had NO idea what "Limit Groups" was... in Deadline you have either "Limits" which does not take a list of machines but predefined limit groups. OR you have a Machine List which is either an allow or deny list. 

![image](https://github.com/user-attachments/assets/258c17ae-01f1-477a-848a-da194c477835)
![image](https://github.com/user-attachments/assets/5f6c96cd-2563-48dc-9145-f9ba939bd84e)
![image](https://github.com/user-attachments/assets/3738862f-75f6-4e6a-9ed0-e987985c337d)

So I think we have the terminology wrong here.

Reply:
Yes, placeholder is wrong.

Reply:
You've updated the placeholder - but I'd actually also relabel them to `Limits` instead of `Limit Groups` to align with Deadline submission UIs.

Reply:
Replaced with enums